test/test_package_descriptor.py

```diff
@@ -33,15 +34,18 @@ def test_identifies_package():
 def test_get_dependencies():
     d1 = PackageDescriptor('/some/path')
     d1.name = 'self'
-    d1.dependencies['build'].add('build-depend')
-    d1.dependencies['build'].add('depend')
-    d1.dependencies['run'].add('run-depend')
-    d1.dependencies['run'].add('depend')
-    assert d1.get_dependencies() == {'build-depend', 'run-depend', 'depend'}
-
-    d1.dependencies['test'].add('self')
+    d1.dependencies['build'].add(DependencyDescriptor('build-depend'))
+    d1.dependencies['build'].add(DependencyDescriptor('depend'))
+    d1.dependencies['run'].add(DependencyDescriptor('run-depend'))
+    d1.dependencies['run'].add(DependencyDescriptor('depend'))
+    # TODO: What if the run and build dependencies have different versions?
```

Comment:
I would say the more strict version constraint should be used when merging. In case of a conflict an `error` message should be logged.

Reply:
Should this logic be on the implementer of the package identification extension point instead?
 
We could then say that in colcon-core there will be no conflicting dependencies.